File: Services/ResultService/ResultService.Api/Domain/Entities/Result.cs
namespace ResultService.Api.Domain.Entities;

public class Result
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public Guid QuizId { get; set; }
    public float Score { get; set; }
    public float MaxPossibleScore { get; set; }
    public int TimeTakenSeconds { get; set; }
    public DateTime CompletedAt { get; set; }
    public required ICollection<ResultAnswer> ResultAnswers { get; set; } = new List<ResultAnswer>();
}

File: frontend/src/models/Result.ts
// Result-related TypeScript models used by services and pages

export interface ResultAnswerDto {
  id: string
  questionId: string
  givenAnswer: string
  pointsAwarded?: number
  isCorrect?: boolean
  explanation?: string
}

export interface ResultDto {
  id: string
  userId: string
  quizId: string
  score: number
  maxPossibleScore: number
  percentageScore: number
  timeTakenSeconds: number
  completedAt: string
  passed: boolean
  passingScore?: number
  answers?: ResultAnswerDto[]
  // Optional fields that may be present in some responses
  quizTitle?: string
}

export interface LeaderboardEntryDto {
  rank: number
  username: string
  score: number
}

export interface UserStatsDto {
  totalQuizzesTaken: number
  averageScore: number
  bestScore: number
  lastTakenAt?: string
}
